Core test suite mode name cleanup

kripken opened the pull request:
* Deprecate running the asm.js tests when the test runner is run with no arguments. That's especially silly now that wasm is the default, but also kind of weird to run any tests with no arguments. Instead, with no arguments, show the help and halt.
 * Rename 'default' to 'asm0', which makes it easy to do `asm*.test_name` for running all asm.js tests.
 * Also deprecate `ALL.` which used to run the asm.js tests. ALL was useful in the past before we had proper wildcard support, but now we can just do `asm*.test_loop` etc.
 * There is still one notion of "default"-ness left, if you do `./tests/runner.py test_loop` then it runs `asm0.test_loop` (same behavior as before). Maybe it makes sense to also switch that to wasm, although I'm not sure which of the wasm modes - seems ok to leave it `asm0` to me, as it feels like the intention there is "run this test somehow".

kripken commented:
Oops, I forgot to actually make the change in the CI config files... done.

kripken commented:
Also noticed that we only warn on tests not found - so `ALL.test_loop` was just a warning, and CI was still green. I turned that into an error.

sbc100 commented on tests/runner.py:
```diff
@@ -134,12 +150,13 @@ def get_browser():
   'asm3',
   'asm2g',
 ]
+default_test_mode = test_modes[0]
```
Shouldn't "binaryen0" be the default_test_mode now?

Likewise shouldn't the asm* modes be non-default?

kripken replied:
Yeah, I wasn't sure about the default mode. But I think you're right, I'll change it to `binaryen0`.

Actually I think we can remove the difference between default and non-default modes - we just need a single default mode for when no mode is specified, but otherwise, that's it. I'll look into that here, I think it'll make things even simpler in the code.

kripken commented:
I simplified the mode handling code - binaryen0 is now the default, and there is no other sense of "default" vs "non-default" modes.

I also realized we have a doc page for the test suite, I moved the docs there and improved them.
